app/views/submissions/steps/trial/_step04_before.html.haml

```diff
@@ -1,54 +1,118 @@
+- pl_pv_name = 'line'
+- pl_pv_name = 'variety' if submission.content.step03.lines_or_varieties == 'plant_varieties'
+- design_factor_names = submission.content.step03.design_factor_names || []
+- technical_replicate_numbers = submission.content.step03.technical_replicate_numbers || {}
+- rep_traits = technical_replicate_numbers.select{ |k,v| v.to_i > 1 }.to_a
+
 .alert.alert-info
+  %h5 Trait scoring sheet structure
   %p
-    Please upload file with score values. The required format is simple and features <b>unique</b> plant
-    identification in the first column and trait scores in column two, three and so forth. The
-    number of columns depends on how many Traits you have chosen in the second step of this submission.
-    All values should be separated by a comma (<code>,</code>).
+    Please upload file with score values. The required format is simple and features
+    %b unique plant scoring unit identification in the first column
+    (please go back to step 3 for short definition of a plant scoring unit), then
+    - if design_factor_names.present?
+      %b #{design_factor_names.size} columns for design factors
+      selected in the previous step,
+    %b 2 columns about plant accession
+    (accession id and accession assigning organisation - see below),
+    %b another column with plant #{pl_pv_name} name
+    (also explained below) and
+    %b trait scores
+    (including potential technical replicate scores) in further columns.
+    The number of columns depends on
+    %ul
+      %li how many <b>Traits</b> you have chosen in the second step of this submission,
+      %li how many <b>design factors</b> you have defined in the third step (raw data only),
+      %li how many <b>technical replicates</b> you plan to submit (raw data only).
+    All values should be separated by a comma (<code>,</code>). In cases where a value is missing,
+    please remember to provide the correct number of commas anyway. Sticking to this template
+    should guarantee correct parsing of score values when you upload the ready file.
 
   %p
-    For instance, if in the second step, you've picked two traits, <i>oleic acid content</i> and
-    <i>seed mature time</i>, you should provide a file like this:
-
-  %pre
-    :preserve
-      Plant id,oleic acid content,seed mature time
-      assay_09,56.98,249.00
-      assay_08,57.24,252.00
-      assay_98,18.48,255.00
-      assay_97,20.40
-      assay_96,50.82,254.00
-      assay_93,58.50
-      assay_94,,252.00
+    For your convenience please use
+    %b
+      = link_to 'the ready template', new_submission_upload_path(submission)
```

Comment:
This link is not that easy to find in the block of text. Maybe we should add a button below the blue text block, before "Choose a file" button?